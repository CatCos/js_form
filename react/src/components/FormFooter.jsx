import React from 'react';
import PropTypes from 'prop-types';

import { FORM_FOOTER } from '../constants/form';

import Input from './Input.jsx';
import Button from './Button.jsx';
/**
 * Render function
 *
 * @params props Component properties
 */
class FormFooter extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      subscribe: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  /**
   * Handle change event
   *
   * @param e event
   */
  handleChange(data) {
    const { e, inputName } = data;

    this.state[inputName] = !this.state[inputName];

    this.setState(this.state);

    if (this.props.onChange) {
      return this.props.onChange({
        inputName: inputName,
        value: this.state[inputName]
      });
    }
  }

  handleSubmit(e) {
    e.preventDefault();

    let value = e.target.value ? e.target.value : e.currentTarget.value;

    if (this.props.onClick) {
      return this.props.onClick({
        value: this.state.subscribe
      });
    }

  }

  renderLeftForm() {
    const fields = FORM_FOOTER.fields;

    let items = fields.map((field, key) => {
      return (
        <Input
          key={key}
          name={field.name}
          value={this.state[field.field]}
          text={field.text}
          type={field.type}
          field={field.field}
          onChange={this.handleChange}
          className='checkbox'/>
      );
    });

    return items;
  }

  renderButton() {
    return (
      <Button
        text='Submit comment'
        onSubmit={this.handleSubmit}/>
    );
  }

  render() {
    return (
      <div className="form-footer">
           <div className="left-form-footer">
              {this.renderLeftForm()}
           </div>
           <div className="right-form-footer">
               {this.renderButton()}
           </div>
       </div>
    );
  }

}

FormFooter.propTypes = {
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired
};

FormFooter.defaultProps = {
};

export default FormFooter;
