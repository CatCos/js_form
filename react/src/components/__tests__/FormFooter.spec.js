import React from 'react';
import FormFooter from '../FormFooter';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <FormFooter
      onClick={jest.fn()}
      onChange={jest.fn()}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
