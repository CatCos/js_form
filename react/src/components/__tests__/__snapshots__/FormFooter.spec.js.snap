// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-footer"
>
  <div
    className="left-form-footer"
  >
    <div
      className="form-group"
    >
      <label
        className="label"
        htmlFor="subscribe"
      >
        Subscribe to follow up comments from this post
      </label>
      <input
        className={
          Array [
            "checkbox",
          ]
        }
        id="subscribe"
        name="subscribe"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
    </div>
  </div>
  <div
    className="right-form-footer"
  >
    <div
      className="form-group"
    >
      <button
        className="submit-button"
        onClick={[Function]}
      >
        Submit comment
      </button>
    </div>
  </div>
</div>
`;
